File: scripts/make_deck.js
const fs = require("fs");
const path = require("path");
const pptxgen = require(path.join(__dirname, "..", "vendor", "pptxgenjs"));

const OUTPUT_DIR = path.join(__dirname, "..", "output");
const OUTPUT_FILE = path.join(OUTPUT_DIR, "antonai_chatgpt_deepdive.pptx");
const OUTPUT_BASE64 = path.join(
  OUTPUT_DIR,
  "antonai_chatgpt_deepdive.pptx.b64.txt"
);

if (!fs.existsSync(OUTPUT_DIR)) {
  fs.mkdirSync(OUTPUT_DIR, { recursive: true });
}

const pptx = new pptxgen();
pptx.layout = "LAYOUT_WIDE";
pptx.author = "AntonAI";
pptx.company = "AntonAI";
pptx.subject = "ChatGPT Deep Dive";
pptx.theme = {
  headFontFace: "Arial",
  bodyFontFace: "Arial",
  lang: "ko-KR"
};

const COLORS = {
  text: "000000",
  accent: "2B6CB0",
  bg: "FFFFFF"
};

const TITLE_STYLE = {
  x: 0.6,
  y: 0.2,
  w: 12.1,
  h: 0.6,
  fontSize: 28,
  bold: true,
  color: COLORS.text
};

const BODY_STYLE = {
  x: 0.9,
  y: 1.4,
  w: 11.6,
  h: 5.4,
  fontSize: 20,
  color: COLORS.text,
  valign: "top"
};

function addHeader(slide, title) {
  slide.background = { color: COLORS.bg };
  slide.addShape(pptx.ShapeType.line, {
    x: 0.6,
    y: 0.95,
    w: 12.1,
    h: 0,
    line: { color: COLORS.accent, width: 1 }
  });
  slide.addText(title, TITLE_STYLE);
}

function addBullets(slide, lines) {
  slide.addText(lines.map((line) => `• ${line}`).join("\n"), BODY_STYLE);
}

const slides = [
  {
    title: "ChatGPT 사용법을 뼛속까지 (안똔AI)",
    subtitle: "Apps / 개발자설정 / MCP / 자동화 / 실전 데모"
  },
  {
    title: "오늘 얻어갈 것 5가지",
    bullets: [
      "ChatGPT 화면 구성 요소를 한 장으로 이해",
      "Apps와 모델의 역할 분담을 구분",
      "개발자 설정으로 생산성 올리기",
      "MCP 연결과 안전 규칙 습득",
      "실전 데모로 자동화 흐름 설계"
    ]
  },
  {
    title: "ChatGPT 화면 구조 한 장 정리",
    bullets: [
      "모델: 답변의 두뇌 (GPT 계열 선택)",
      "툴: 파일, 브라우저, 코드 실행 등 확장",
      "앱: 외부 서비스와 연결되는 UI",
      "파일: 참고 자료/근거를 직접 넣는 방식",
      "프로젝트: 작업 컨텍스트와 히스토리 묶음"
    ]
  },
  {
    title: "Apps: 앱이 하는 일 vs 모델이 하는 일",
    bullets: [
      "앱: Canva, Notion 등 UI/데이터 연결",
      "모델: 요약/분석/생성 같은 사고 작업",
      "앱은 권한/데이터를 가져오고",
      "모델은 지시문을 따라 결과를 만들기",
      "둘의 경계를 이해하면 결과가 안정적"
    ]
  },
  {
    title: "개발자 설정: 뭘 켜면 뭐가 달라질까",
    bullets: [
      "도구 허용: 모델이 외부 기능 호출",
      "파일 처리: PDF/CSV 분석 속도 개선",
      "메모리/프로젝트: 장기 컨텍스트 유지",
      "실험 기능: 베타 기능 빠르게 체험",
      "권한 설정: 안전 범위 명확히 하기"
    ]
  },
  {
    title: "MCP: 개념 + 안전하게 쓰는 룰",
    bullets: [
      "MCP = 모델이 도구를 호출하는 연결 규칙",
      "서버/클라이언트로 기능 확장",
      "권한 최소화: 꼭 필요한 도구만 연결",
      "민감 데이터는 로컬/폐쇄망 우선",
      "로그/감사로 호출 내역 확인"
    ]
  },
  {
    title: "실전 데모 1: 문서/표/리서치 3-step",
    bullets: [
      "1) 자료 수집: 파일 업로드 + 요약",
      "2) 구조화: 표/목차로 재정리",
      "3) 인사이트: 비교/결론/다음 액션",
      "템플릿화하면 재사용이 쉬움"
    ]
  },
  {
    title: "실전 데모 2: 자동화 시나리오",
    bullets: [
      "조건: 어떤 이벤트가 시작 트리거인지",
      "트리거: 스케줄/웹훅/파일 변경",
      "출력: 문서/보고서/알림으로 자동 발행",
      "실패 대비: 재시도와 에러 로그",
      "작게 시작해서 점진적으로 확장"
    ]
  },
  {
    title: "Codex란?",
    bullets: [
      "ChatGPT 채팅: 대화 중심",
      "Codex 작업공간: 프로젝트/파일 중심",
      "여러 파일 수정 + 실행 + 검증까지",
      "팀/레포 단위로 협업 가능",
      "장기 작업에 최적화된 워크플로"
    ]
  },
  {
    title: "Codex로 산출물 만들기: 끝까지",
    bullets: [
      "코드 작성 → 실행 → 결과 확인",
      "문서/슬라이드 자동 생성",
      "버전 관리와 변경 이력 유지",
      "오늘 실습: 슬라이드 덱 자동 생성",
      "반복 작업을 파이프라인화"
    ]
  },
  {
    title: "체크리스트: 실패 vs 성공 지시문",
    bullets: [
      "실패: 목적이 모호하고 예시가 없음",
      "실패: 형식/길이/제약 조건 누락",
      "성공: 목표 + 결과물 + 제약 명확",
      "성공: 입력 데이터와 맥락 제공",
      "성공: 검수 기준(체크리스트) 포함"
    ]
  },
  {
    title: "엔딩",
    bullets: [
      "다음 영상: MCP 실전 자동화 구축기",
      "구독/좋아요/댓글로 질문 남기기",
      "자료 링크 자리: QR 또는 URL",
      "오늘의 핵심: 구조 → 자동화 → 검증"
    ]
  }
];

slides.forEach((data, index) => {
  const slide = pptx.addSlide();
  addHeader(slide, data.title);

  if (index === 0 && data.subtitle) {
    slide.addText(data.subtitle, {
      x: 0.9,
      y: 2.1,
      w: 11.6,
      h: 1.2,
      fontSize: 22,
      color: COLORS.text
    });
  } else if (data.bullets) {
    addBullets(slide, data.bullets);
  }
});

pptx.writeFile({ fileName: OUTPUT_FILE }).then(() => {
  const deckBinary = fs.readFileSync(OUTPUT_FILE);
  fs.writeFileSync(OUTPUT_BASE64, deckBinary.toString("base64"));
  console.log(`Deck created: ${OUTPUT_FILE}`);
  console.log(`Base64 created: ${OUTPUT_BASE64}`);
});

File: scripts/make_deck.py
from __future__ import annotations

from pathlib import Path

from pptx import Presentation
from pptx.dml.color import RGBColor
from pptx.enum.shapes import MSO_AUTO_SHAPE_TYPE
from pptx.util import Inches, Pt


OUTPUT_DIR = Path(__file__).resolve().parent.parent / "output"
OUTPUT_FILE = OUTPUT_DIR / "antonai_chatgpt_deepdive.pptx"

ACCENT = RGBColor(0x1E, 0x88, 0xE5)
BLACK = RGBColor(0x00, 0x00, 0x00)

SLIDES = [
    {
        "title": "ChatGPT 사용법을 뼛속까지 (안똔AI)",
        "lines": ["Apps / 개발자설정 / MCP / 자동화 / 실전 데모"],
        "bullets": False,
    },
    {
        "title": "오늘 얻어갈 것 5가지",
        "lines": [
            "ChatGPT 핵심 화면 구조 이해",
            "앱과 모델의 역할 분리",
            "개발자 설정으로 품질·안전 강화",
            "MCP로 도구 연결하는 법",
            "실전 데모 2개 워크플로",
        ],
        "bullets": True,
    },
    {
        "title": "ChatGPT 화면 구조",
        "lines": [
            "모델: 답변을 만드는 두뇌",
            "툴: 계산/검색/실행 같은 손발",
            "앱: 캔버스·캘린더 등 연결 서비스",
            "파일: 입력/출력 자료",
            "프로젝트: 목적별 지식·맥락 저장소",
        ],
        "bullets": True,
    },
    {
        "title": "Apps: Canva 포함",
        "lines": [
            "앱이 하는 일: UI 제공, 외부 서비스 접속",
            "모델이 하는 일: 지시 해석, 콘텐츠 생성",
            "앱은 ‘창구’, 모델은 ‘엔진’",
            "연결 앱 목록은 사용 시점에만 권한 부여",
        ],
        "bullets": True,
    },
    {
        "title": "개발자 설정(Developer settings)",
        "lines": [
            "개발자 모드: 도구/프로젝트 권한 제어",
            "안전 필터: 위험한 호출 차단",
            "실험 기능: 최신 모델/툴 우선 사용",
            "로그 확인: 실패 원인 추적",
        ],
        "bullets": True,
    },
    {
        "title": "MCP 핵심 개념",
        "lines": [
            "도구 호출을 표준화하는 연결 프로토콜",
            "모델이 ‘요청’, 서버가 ‘실행’",
            "권한 최소화 + 읽기 전용부터 시작",
            "환경 분리(개발/운영)로 사고 방지",
        ],
        "bullets": True,
    },
    {
        "title": "실전 데모 1: 문서/표/리서치",
        "lines": [
            "1) 요구사항 요약 → 질문 리스트 생성",
            "2) 문서·표·리서치 병렬 수집",
            "3) 핵심 인사이트 5줄로 정리",
        ],
        "bullets": True,
    },
    {
        "title": "실전 데모 2: 자동화 시나리오",
        "lines": [
            "조건: 신규 파일 업로드 감지",
            "트리거: 요약 + 분류 + 알림",
            "출력: 슬랙 메시지 + 보고서 초안",
        ],
        "bullets": True,
    },
    {
        "title": "Codex란?",
        "lines": [
            "ChatGPT 채팅: 대화 중심",
            "Codex 작업공간: 파일/코드 중심",
            "버전 관리 + 자동 실행",
            "다단계 산출물에 최적화",
        ],
        "bullets": True,
    },
    {
        "title": "Codex로 산출물 만들기",
        "lines": [
            "코드 → 문서 → 슬라이드까지 연결",
            "목표·제약·출력물 3줄로 정의",
            "반복 피드백으로 ‘끝까지’ 완주",
            "오늘 실습: 슬라이드 자동 생성",
        ],
        "bullets": True,
    },
    {
        "title": "체크리스트",
        "lines": [
            "실패: 모호한 목적, 범위 없음",
            "실패: 출력 형식 지시 누락",
            "성공: 목적/대상/형식/제약 명시",
            "성공: 예시 제공 + 검증 기준",
            "성공: 단계별 확인 요청",
        ],
        "bullets": True,
    },
    {
        "title": "엔딩",
        "lines": [
            "다음 영상: MCP 실전 서버 만들기",
            "구독/자료 링크 자리",
            "질문은 댓글에!",
        ],
        "bullets": True,
    },
]


def build_deck() -> None:
    presentation = Presentation()
    presentation.slide_width = Inches(13.333)
    presentation.slide_height = Inches(7.5)

    blank_layout = presentation.slide_layouts[6]
    title_left = Inches(0.7)
    title_top = Inches(0.3)
    title_width = Inches(12.0)
    title_height = Inches(0.6)

    line_left = Inches(0.7)
    line_top = Inches(0.98)
    line_width = Inches(12.0)
    line_height = Inches(0.04)

    body_left = Inches(0.9)
    body_top = Inches(1.3)
    body_width = Inches(11.6)
    body_height = Inches(5.6)

    for slide_data in SLIDES:
        slide = presentation.slides.add_slide(blank_layout)
        title_box = slide.shapes.add_textbox(
            title_left, title_top, title_width, title_height
        )
        title_frame = title_box.text_frame
        title_frame.clear()
        title_frame.text = slide_data["title"]
        title_run = title_frame.paragraphs[0].runs[0]
        title_run.font.size = Pt(32)
        title_run.font.bold = True
        title_run.font.color.rgb = BLACK

        line = slide.shapes.add_shape(
            MSO_AUTO_SHAPE_TYPE.RECTANGLE,
            line_left,
            line_top,
            line_width,
            line_height,
        )
        line.fill.solid()
        line.fill.fore_color.rgb = ACCENT
        line.line.fill.background()

        body_box = slide.shapes.add_textbox(
            body_left, body_top, body_width, body_height
        )
        body_frame = body_box.text_frame
        body_frame.clear()
        for idx, line_text in enumerate(slide_data["lines"]):
            paragraph = body_frame.paragraphs[0] if idx == 0 else body_frame.add_paragraph()
            paragraph.text = line_text
            paragraph.level = 0
            paragraph.font.size = Pt(20)
            paragraph.font.color.rgb = BLACK
            paragraph.bullet = slide_data["bullets"]

    OUTPUT_DIR.mkdir(parents=True, exist_ok=True)
    presentation.save(OUTPUT_FILE)


if __name__ == "__main__":
    build_deck()
    print(f"Generated: {OUTPUT_FILE}")
